src/worker/tasks/processTx.ts

```diff
@@ -178,27 +178,55 @@ export const processTx = async () => {
           }
 
           // Group all transactions into a single batch rpc request
+          let i = 0;
           const rpcRequests = [];
-          for (const i in txsToSend) {
-            const tx = txsToSend[i];
+          const preparedTxs: typeof txsToSend = [];
+          for (const tx of txsToSend) {
             const nonce = startNonce.add(i);
 
-            const txRequest = await signer.populateTransaction({
-              to: tx.toAddress!,
-              from: tx.fromAddress!,
-              data: tx.data!,
-              value: tx.value!,
-              nonce,
-              ...gasOverrides,
-            });
-            const signature = await signer.signTransaction(txRequest);
-
-            rpcRequests.push({
-              id: i,
-              jsonrpc: "2.0",
-              method: "eth_sendRawTransaction",
-              params: [signature],
-            });
+            try {
+              const txRequest = await signer.populateTransaction({
+                to: tx.toAddress!,
+                from: tx.fromAddress!,
+                data: tx.data!,
+                value: tx.value!,
+                nonce,
+                ...gasOverrides,
+              });
+              const signature = await signer.signTransaction(txRequest);
+
+              rpcRequests.push({
+                id: i,
+                jsonrpc: "2.0",
+                method: "eth_sendRawTransaction",
+                params: [signature],
+              });
+
+              preparedTxs.push(tx);
+
+              i++;
+            } catch (err: any) {
+              logger.worker.warn(
+                `[Transaction] [${
+                  tx.queueId
+                }] Failed to send transaction with error - ${
+                  err?.message || err
+                }`,
+              );
+
+              await updateTx({
```

Comment:
can this fail?

Reply:
In what scenario? This is just making a raw call to postgres via prisma

Reply:
We're using it below in this file too